tests/unit/features/partitions/test_parts.py

```diff
@@ -130,39 +130,94 @@ def valid_filesets(self):
         return [
             "test",
             "(default)",
+            "(default)/",
+            "(default)//",
             "(default)/test",
+            "(default)//test",
             "(default)/test/(otherdir)",
             "(default)/test/(otherdir-123)",
             "(default)/test/(default)",
             "(default)/test/(kernel)",
+            "(default)/test/(a/b)",
+            "(default)/test/(a/c-d)",
             "test(default)",
             "test(default)/test",
             "test/(default)/test",
             "(kernel)",
+            "(kernel)/",
+            "(kernel)//",
             "(kernel)/test",
+            "(kernel)//test",
             "(kernel)/test/(otherdir)",
             "(kernel)/test/(otherdir-123)",
             "(kernel)/test/(default)",
             "(kernel)/test/(kernel)",
+            "(kernel)/test/(a/b)",
+            "(kernel)/test/(a/c-d)",
             "test/(kernel)",
             "test(kernel)/test",
+            "test/(kernel)/test",
+            "(a/b)",
+            "(a/b)/",
+            "(a/b)//",
+            "(a/b)/test",
+            "(a/b)//test",
+            "(a/b)/test/(otherdir)",
+            "(a/b)/test/(otherdir-123)",
+            "(a/b)/test/(default)",
+            "(a/b)/test/(kernel)",
+            "(a/b)/test/(a/b)",
+            "(a/b)/test/(a/c-d)",
+            "test/(a/b)",
+            "test(a/b)/test",
+            "test/(a/b)/test",
+            "(a/c-d)",
+            "(a/c-d)/",
+            "(a/c-d)//",
+            "(a/c-d)/test",
+            "(a/c-d)//test",
+            "(a/c-d)/test/(otherdir)",
+            "(a/c-d)/test/(otherdir-123)",
+            "(a/c-d)/test/(default)",
+            "(a/c-d)/test/(kernel)",
+            "(a/c-d)/test/(a/b)",
+            "(a/c-d)/test/(a/c-d)",
+            "test/(a/c-d)",
+            "test(a/c-d)/test",
+            "test/(a/c-d)/test",
         ]
 
     @pytest.fixture()
     def invalid_filesets(self):
         """Return a list of invalid filesets when the partition feature is enabled.
 
-        Assumes "default" and "partition" are the only valid partitions.
+        Assumes "default", "a/b", "a/c-d" and "kernel" are the only valid partitions.
         """
         return [
             "()",
             "(foo)",
             "(bar)/test",
-            "(baz)",
+            "(BAZ)",
+            "(foo)/",
```

Comment:
I see that `foo` is not a valid partition, why is `(foo)/` it invalid when a similar `(kernel)/` in the previous list is valid?

Reply:
This is a confusing name for a fixture.  

I'll update the docstring. These filepaths are not necessarily violating the naming convention for partitions but the partitions here are unknown (and thus invalid) to a LifecycleManager was created with the partitions `default`, `a/b`, `a/c-d`, and `kernel`.